docs/source/settings.rst

```diff
@@ -359,6 +367,11 @@ speed up server boot times. Although, if you defer application loading
 to each worker process, you can reload your application code easily by
 restarting workers.
 
+.. note::
```

Comment:
maybe being more generic and only making gevent and eventlet as examples?

Reply:
I'm happy to change the wording but both gevent and eventlet advise calling `monkey_patch()` functions early as possible (and calling them in module level (in `project/__init__.py` for example) is one of the ways to do it) What is the use cause of calling those functions later?

Reply:
I can mention that a large gevent application we deploy *must* use `preload-app` due to a heavy startup time, and in order for that to work we *must* monkey-patch the entire gunicorn arbiter process, which we do by calling `gevent.monkey.patch_all()` in a script before loading and invoking gunicorn's entry point.

Also, gevent does strongly recommend monkey-patching ASAP and that very often means in a module-global location. That's common during development, when a module might look like:
```python
from gevent import monkey
monkey.patch_all()

from myproject.app import wsgi_app # safely separated to another module away from the monkey
from gevent.pywsgi import WSGIServer

server = WSGIServer(('localhost', 0), wsgi_app)
server.serve_forever()
```

But it's not unusual for it to also look like this:
```python
from gevent import monkey
monkey.patch_all()

from gevent.pywsgi import WSGIServer

def wsgi_app(environ, start_response):
   "We can be used with gunicorn or in development."
   "Note how we're in the same module as our monkey patch"
   
if __name__ == '__main__':
    server = WSGIServer(('localhost', 0), wsgi_app)
    server.serve_forever()
```

Reply:
If we want to document using a custom application [1] for preloading with gevent/eventlet, I would be fine with that. Otherwise, enough things have been imported by the time gunicorn reads the configuration that I'm not sure we can monkey patch early enough. It might be interesting to try, though, by seeing how much we can factor out the loading steps into different modules. At this point, though, without a custom application I don't think we can advise using preload with gevent.

[1] http://docs.gunicorn.org/en/latest/custom.html